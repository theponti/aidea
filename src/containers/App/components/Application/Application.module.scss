.loading {
  text-align: center;
  text-transform: uppercase;
  font-size: 1.5rem;
  font-weight: 800;
  padding: 10rem 0;
}