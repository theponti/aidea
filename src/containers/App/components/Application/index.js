import { useAuth0 } from '@auth0/auth0-react'
import React from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import Home from 'src/views/Home'
import Ideas from 'src/views/Ideas'
import ProfilePage from 'src/views/ProfilePage'
import styles from './Application.module.scss'

function Application () {
  const { isLoading, error } = useAuth0()

  if (isLoading) {
    return <h2 className={styles.loading}>Loading...</h2>
  }

  if (error) {
    return (
      <div>
        <h2>Error</h2>
        <p>{error.message}</p>
      </div>
    )
  }

  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/profile" component={ProfilePage} />
        <Route path="/ideas" component={Ideas} />
      </Switch>
    </Router>
  )
}

export default Application
