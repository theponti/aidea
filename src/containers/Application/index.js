import { useAuth0 } from '@auth0/auth0-react'
import { Router } from '@reach/router'
import React from 'react'
import Home from 'src/views/Home'
import Ideas from 'src/views/Ideas'
import ProfilePage from '../ProfilePage'
import styles from './Application.module.scss'

function Application () {
  const { isLoading, error } = useAuth0()

  if (isLoading) {
    return <h2 className={styles.loading}>Loading...</h2>
  }

  if (error) {
    return (
      <div>
        <h2>Error</h2>
        <p>{error.message}</p>
      </div>
    )
  }

  return (
    <Router>
      <Home default path="/" />
      <ProfilePage path="profile" />
      <Ideas path="ideas" />
    </Router>
  )
}

export default Application
