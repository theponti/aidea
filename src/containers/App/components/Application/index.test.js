import { render } from '@testing-library/react'
import React from 'react'
import Application from '.'

const { useAuth0 } = require('@auth0/auth0-react')

describe('', () => {
  it('should render loading if authenticated is false', () => {
    useAuth0.mockReturnValue({ isLoading: true })
    const { getByText } = render(<Application />)
    expect(getByText(/loading/i)).toBeInTheDocument()
  })

  it('should render error', () => {
    useAuth0.mockReturnValue({ error: { message: 'error' } })
    const { getByText } = render(<Application />)
    expect(getByText(/error/)).toMatchSnapshot()
  })
})
