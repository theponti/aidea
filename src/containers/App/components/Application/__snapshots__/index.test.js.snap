// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render error 1`] = `
<p>
  error
</p>
`;
